recipes/magnum/all/test_package/CMakeLists.txt

```diff
@@ -26,6 +26,14 @@ set_target_properties(${PROJECT_NAME}
         CXX_EXTENSIONS OFF
 )
 
+if(TARGET_EMSCRIPTEN)
+    message("Embed OBJ file")
+    set_target_properties(${PROJECT_NAME} 
+        PROPERTIES 
+            LINK_FLAGS "--embed-file ${OBJ_FILE}@${OBJ_FILE}"
```

Comment:
what exactly does it do and why is it needed?

Reply:
This embeds a file into the generated `js`. The generated _application_ cannot access the filesystem, but it can use a virtual filesystem with the embedded files: https://emscripten.org/docs/porting/files/packaging_files.html

Reply:
okay, I was confused by OBJ_FILE. I was thinking it's about `.obj` object files produced by compiler (linkable object).
but it seems to be test .obj file, which is 3d model format ([wavefront obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file)).
it's okay to embed test file in this case.